frontend/model/captureLogs.js

```diff
@@ -0,0 +1,186 @@
+import sbp from '~/shared/sbp.js'
+import { CAPTURED_LOGS, SET_APP_LOGS_FILTER } from '~/frontend/utils/events.js'
+
+/*
+  giConsole/count - total logs stored
+  giConsole/lastEntry - latest log hash.
+  giConsole/markers - an array of markers at every 1000th entry.
+*/
+const ENTRIES_LIMIT = 4000
+const ENTRIES_MARKER = 1000
+
+let isCapturing = true
+let isConsoleOveride = false
+let lastEntry = null
+let entriesCount = null
+let appLogsFilter = []
+
+function hashFn () {
+  // REVIEW - a better hash?
+  return `${Date.now()}_${Math.floor(Math.random() * 100000)}`
+}
+
+function captureLog (type, ...msg) {
+  const logEntry = hashFn()
+
+  localStorage.setItem(`giConsole/${logEntry}`,
+    JSON.stringify({ type, msg, prev: lastEntry })
+  )
+
+  lastEntry = logEntry
+  entriesCount++
+
+  localStorage.setItem('giConsole/lastEntry', logEntry)
+  localStorage.setItem('giConsole/count', entriesCount)
+
+  verifyLogsSize()
+
+  sbp('okTurtles.events/emit', CAPTURED_LOGS, lastEntry)
+}
+
+function verifyLogsSize () {
+  if (entriesCount % ENTRIES_MARKER === 0) {
+    // Save entry as a marker to be later deleted when logs are too big.
+    const markers = JSON.parse(localStorage.getItem('giConsole/markers')) || []
+    markers.push(lastEntry)
+    localStorage.setItem('giConsole/markers', JSON.stringify(markers))
+  }
+
+  if (entriesCount >= ENTRIES_LIMIT) {
+    const markers = JSON.parse(localStorage.getItem('giConsole/markers')) || []
```

Comment:
DRY (see line 44), just have a single line at the top of this function for this.

Reply:
With that approach, you write less code but you execute it more than needed. this way the code is only executed when a if statement is valid, instead of everytime.